### PR TITLE
Fixes for 12.2 and 18.3

### PR DESCRIPTION
**Pull request by psaini79**
I had some fixes for 12.2 and 18.3 based on user input. Also, I have changes for Anisble.

### CODE REVIEW

File: OracleDatabase/RAC/OracleRealApplicationClusters/dockerfiles/12.2.0.1/MultiNodeInstall.py [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+
+# stig.py
+#
+# Copyright (c) 2012, 2018, Oracle and/or its affiliates. All rights reserved.
+#
+# NAME
+#      buildImage.py - <Build the Image>
+#
+# DESCRIPTION
+#      <This  script will build the Image>
```
Comment (Djelibeybi) [inline OracleDatabase/RAC/OracleRealApplicationClusters/dockerfiles/12.2.0.1/MultiNodeInstall.py]: This is not a great description. What image? What build? What are the parameters? If you're going to have a description, please make it descriptive. I see a bunch of parameters and options lower in the script, so those should be documented here as well.

File: OracleDatabase/RAC/OracleRealApplicationClusters/dockerfiles/12.2.0.1/setupGrid.sh [hunk 2]
```diff
@@ -26,12 +26,10 @@ chown -R grid:oinstall $GRID_HOME
...
 chown -R grid:oinstall $INVENTORY
 
-############# Disabled sudo Access ######
-#### However, if user need the sudo access for his env, he can remove the comments from following lines and end rebuild the image ###############
-#chmod 666 /etc/sudoers
-#echo "oracle       ALL=(ALL)       NOPASSWD: ALL" >> /etc/sudoers
-#chmod 440 /etc/sudoers
-########## sudo code end here ###########
+chmod 666 /etc/sudoers
+echo "oracle       ALL=(ALL)       NOPASSWD: ALL" >> /etc/sudoers
```
Comment (Djelibeybi) [inline OracleDatabase/RAC/OracleRealApplicationClusters/dockerfiles/12.2.0.1/setupGrid.sh]: Do you really need to grant `oracle` full root access?

File: OracleDatabase/RAC/OracleRealApplicationClusters/dockerfiles/12.2.0.1/setupGrid.sh [hunk 3]
```diff
@@ -26,12 +26,10 @@ chown -R grid:oinstall $GRID_HOME
...
+echo "grid       ALL=(ALL)       NOPASSWD: ALL" >> /etc/sudoers
```
Comment (Djelibeybi) [inline OracleDatabase/RAC/OracleRealApplicationClusters/dockerfiles/12.2.0.1/setupGrid.sh]: Same here. Does the `grid` user need full root access?

File: OracleDatabase/RAC/OracleRealApplicationClusters/dockerfiles/12.2.0.1/setupLinuxEnv.sh [hunk 4]
```diff
@@ -16,5 +16,5 @@
 mkdir /asmdisks && \
 mkdir /responsefiles  && \
 chmod ug+x /opt/scripts/startup/*.sh && \
-yum -y install systemd oracle-database-server-12cR2-preinstall net-tools which zip unzip tar openssl expect e2fsprogs openssh-server openssh-client vim-minimal passwd which sudo && \
+yum -y install systemd  oracle-database-server-12cR2-preinstall  net-tools which zip unzip tar openssl expect e2fsprogs openssh-server openssh-client vim-minimal passwd which sudo xorg-x11-server-Xorg xorg-x11-xauth xorg-x11-apps && \
```
Comment (Djelibeybi) [inline OracleDatabase/RAC/OracleRealApplicationClusters/dockerfiles/12.2.0.1/setupLinuxEnv.sh]: What's the reason behind adding the `xorg-*` RPMs into a container? I'm not sure trying to start a graphical environment is a good idea here.

File: OracleDatabase/RAC/OracleRealApplicationClusters/dockerfiles/18.3.0/MultiNodeInstall.py [hunk 5]
```diff
@@ -0,0 +1,310 @@
+#!/usr/bin/python
+#!/usr/bin/env python
+
```
Comment (Djelibeybi) [inline OracleDatabase/RAC/OracleRealApplicationClusters/dockerfiles/18.3.0/MultiNodeInstall.py]: Only one shebang line should be necessary.

File: OracleDatabase/RAC/OracleRealApplicationClusters/dockerfiles/12.2.0.1/MultiNodeInstall.py [hunk 6]
```diff
@@ -0,0 +1,310 @@
+#!/usr/bin/python
+#!/usr/bin/env python
+
+###########################################################################################################
+
+
+# stig.py
```
Comment (Djelibeybi) [inline OracleDatabase/RAC/OracleRealApplicationClusters/dockerfiles/12.2.0.1/MultiNodeInstall.py]: The file name is `MultiNodeInstall.py`.

File: OracleDatabase/RAC/OracleRealApplicationClusters/dockerfiles/12.2.0.1/MultiNodeInstall.py [hunk 7]
```diff
@@ -0,0 +1,310 @@
...
+#
+# Copyright (c) 2012, 2018, Oracle and/or its affiliates. All rights reserved.
+#
+# NAME
+#      buildImage.py - <Build the Image>
```
Comment (Djelibeybi) [inline OracleDatabase/RAC/OracleRealApplicationClusters/dockerfiles/12.2.0.1/MultiNodeInstall.py]: The file name is `MultiNodeInstall.py`.

File: OracleDatabase/RAC/OracleRealApplicationClusters/dockerfiles/18.3.0/MultiNodeInstall.py [hunk 8]
```diff
@@ -0,0 +1,310 @@
...
+###########################################################################################################
+
+
+# stig.py
```
Comment (Djelibeybi) [inline OracleDatabase/RAC/OracleRealApplicationClusters/dockerfiles/18.3.0/MultiNodeInstall.py]: The file name is `MultiNodeInstall.py`.

File: OracleDatabase/RAC/OracleRealApplicationClusters/dockerfiles/18.3.0/MultiNodeInstall.py [hunk 9]
```diff
@@ -0,0 +1,310 @@
...
+#
+# Copyright (c) 2012, 2018, Oracle and/or its affiliates. All rights reserved.
```
Comment (Djelibeybi) [inline OracleDatabase/RAC/OracleRealApplicationClusters/dockerfiles/18.3.0/MultiNodeInstall.py]: This file needs at least the 1-line UPL header.

File: OracleDatabase/RAC/OracleRealApplicationClusters/dockerfiles/18.3.0/MultiNodeInstall.py [hunk 10]
```diff
@@ -0,0 +1,310 @@
...
+#
+# NAME
+#      buildImage.py - <Build the Image>
```
Comment (Djelibeybi) [inline OracleDatabase/RAC/OracleRealApplicationClusters/dockerfiles/18.3.0/MultiNodeInstall.py]: The file name is `MultiNodeInstall.py`.

File: OracleDatabase/RAC/OracleRealApplicationClusters/dockerfiles/18.3.0/MultiNodeInstall.py [hunk 11]
```diff
@@ -0,0 +1,310 @@
...
+#
+# DESCRIPTION
+#      <This  script will build the Image>
```
Comment (Djelibeybi) [inline OracleDatabase/RAC/OracleRealApplicationClusters/dockerfiles/18.3.0/MultiNodeInstall.py]: This description is insufficient. Basic instructions on how to call the script and what each parameter controls would be great.

File: OracleDatabase/RAC/OracleRealApplicationClusters/dockerfiles/18.3.0/MultiRACInstall.sh [hunk 12]
```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
```
Comment (Djelibeybi) [inline OracleDatabase/RAC/OracleRealApplicationClusters/dockerfiles/18.3.0/MultiRACInstall.sh]: This script needs both a copyright and a 1-line UPL license header.

File: OracleDatabase/RAC/OracleRealApplicationClusters/dockerfiles/12.2.0.1/MultiRACInstall.sh [hunk 13]
```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
```
Comment (Djelibeybi) [inline OracleDatabase/RAC/OracleRealApplicationClusters/dockerfiles/12.2.0.1/MultiRACInstall.sh]: This script needs both a copyright and a 1-line UPL license header.

File: OracleDatabase/RAC/OracleRealApplicationClusters/dockerfiles/18.3.0/resetOSPassword.sh [hunk 14]
```diff
@@ -58,8 +60,12 @@ echo  $PASSWORD
 }
```
Comment (mriccell) [inline OracleDatabase/RAC/OracleRealApplicationClusters/dockerfiles/18.3.0/resetOSPassword.sh]: echo $PASSWORD should be commented out or only enabled when debugging

File: OracleDatabase/RAC/OracleRealApplicationClusters/dockerfiles/12.2.0.1/resetOSPassword.sh [hunk 15]
```diff
@@ -0,0 +1,214 @@
...
+rm -f  ${SECRET_VOLUME}/${COMMON_OS_PWD_FILE}
+rm -f ${SECRET_VOLUME}/${PWD_KEY}
+fi
+
+else
+ print_message "Password file or password key file is empty string! generating random password"
+ PASSWORD=O$(openssl rand -base64 6 | tr -d "=+/")_1
+fi
+
+echo  $PASSWORD
```
Comment (mriccell) [inline OracleDatabase/RAC/OracleRealApplicationClusters/dockerfiles/12.2.0.1/resetOSPassword.sh]: This echo should be commented out or only enabled when debugging

File: OracleDatabase/RAC/OracleRealApplicationClusters/dockerfiles/12.2.0.1/MultiNodeInstall.py [hunk 16]
```diff
@@ -0,0 +1,310 @@
+#!/usr/bin/python
+#!/usr/bin/env python
```
[the remark this answers is not in the document]
Comment (psaini79) [reply OracleDatabase/RAC/OracleRealApplicationClusters/dockerfiles/12.2.0.1/MultiNodeInstall.py]: Yes, I made the changes.

File: OracleDatabase/RAC/OracleRealApplicationClusters/dockerfiles/12.2.0.1/MultiNodeInstall.py [hunk 17]
```diff
@@ -0,0 +1,310 @@
...
+
+###########################################################################################################
+
+
+# stig.py
+#
+# Copyright (c) 2012, 2018, Oracle and/or its affiliates. All rights reserved.
```
[the remark this answers is not in the document]
Comment (psaini79) [reply OracleDatabase/RAC/OracleRealApplicationClusters/dockerfiles/12.2.0.1/MultiNodeInstall.py]: Thanks. I appreciate and  make the required changes.

File: OracleDatabase/RAC/OracleRealApplicationClusters/dockerfiles/12.2.0.1/MultiNodeInstall.py (hunk 1 above)
Comment (psaini79) [reply OracleDatabase/RAC/OracleRealApplicationClusters/dockerfiles/12.2.0.1/MultiNodeInstall.py]: Added more details

File: OracleDatabase/RAC/OracleRealApplicationClusters/dockerfiles/12.2.0.1/setupGrid.sh (hunk 2 above)
Comment (psaini79) [reply OracleDatabase/RAC/OracleRealApplicationClusters/dockerfiles/12.2.0.1/setupGrid.sh]: Remove the Access for Oracle. However, I need access for Grid as default login user is grid.

File: OracleDatabase/RAC/OracleRealApplicationClusters/dockerfiles/12.2.0.1/setupGrid.sh (hunk 3 above)
Comment (psaini79) [reply OracleDatabase/RAC/OracleRealApplicationClusters/dockerfiles/12.2.0.1/setupGrid.sh]: Removed the access for Oracle.

File: OracleDatabase/RAC/OracleRealApplicationClusters/dockerfiles/18.3.0/MultiNodeInstall.py (hunk 9 above)
Comment (psaini79) [reply OracleDatabase/RAC/OracleRealApplicationClusters/dockerfiles/18.3.0/MultiNodeInstall.py]: Added the details.

File: OracleDatabase/RAC/OracleRealApplicationClusters/dockerfiles/18.3.0/MultiNodeInstall.py (hunk 10 above)
Comment (psaini79) [reply OracleDatabase/RAC/OracleRealApplicationClusters/dockerfiles/18.3.0/MultiNodeInstall.py]: Added the details

File: OracleDatabase/RAC/OracleRealApplicationClusters/dockerfiles/18.3.0/MultiNodeInstall.py (hunk 11 above)
Comment (psaini79) [reply OracleDatabase/RAC/OracleRealApplicationClusters/dockerfiles/18.3.0/MultiNodeInstall.py]: Add the details

File: OracleDatabase/RAC/OracleRealApplicationClusters/dockerfiles/12.2.0.1/MultiRACInstall.sh (hunk 13 above)
Comment (psaini79) [reply OracleDatabase/RAC/OracleRealApplicationClusters/dockerfiles/12.2.0.1/MultiRACInstall.sh]: Removed the file as at this moment it is not required.

File: OracleDatabase/RAC/OracleRealApplicationClusters/dockerfiles/12.2.0.1/DelNode.sh [hunk 18]
```diff
@@ -12,6 +12,7 @@
 declare -a cluster_nodes
 DEL_NODE=${1}
 NODE_HOSTNAME=$(hostname)
+node_count=0
```
[the remark this answers is not in the document]
Comment (psaini79) [reply OracleDatabase/RAC/OracleRealApplicationClusters/dockerfiles/12.2.0.1/DelNode.sh]: Made the required changes.

File: OracleDatabase/RAC/OracleRealApplicationClusters/dockerfiles/12.2.0.1/resetOSPassword.sh [hunk 19]
```diff
@@ -0,0 +1,214 @@
+#!/bin/bash
+# LICENSE UPL 1.0
+#
+# Copyright (c) 1982-2018 Oracle and/or its affiliates. All rights reserved.
```
[the remark this answers is not in the document]
Comment (psaini79) [reply OracleDatabase/RAC/OracleRealApplicationClusters/dockerfiles/12.2.0.1/resetOSPassword.sh]: Made the required changes

File: OracleDatabase/RAC/OracleRealApplicationClusters/dockerfiles/18.3.0/resetOSPassword.sh (hunk 14 above)
Comment (psaini79) [reply OracleDatabase/RAC/OracleRealApplicationClusters/dockerfiles/18.3.0/resetOSPassword.sh]: Thanks. I fixed it.

File: OracleDatabase/RAC/OracleRealApplicationClusters/dockerfiles/12.2.0.1/resetOSPassword.sh (hunk 15 above)
Comment (psaini79) [reply OracleDatabase/RAC/OracleRealApplicationClusters/dockerfiles/12.2.0.1/resetOSPassword.sh]: Fixed it.

File: OracleDatabase/RAC/OracleRealApplicationClusters/dockerfiles/12.2.0.1/AddNode.sh [hunk 20]
```diff
@@ -575,50 +733,50 @@ local vip_hostname=$VIP_HOSTNAME
...
+  IFS=', ' read -r -a CLUSTER_NODES <<< "$CRS_NODES"
+fi
+
 if [ -f "$logdir/cluvfy_check.txt" ]; then
 print_message "Moving any exisiting cluvfy $logdir/cluvfy_check.txt to $logdir/cluvfy_check_$TIMESTAMP.txt"
 mv $logdir/cluvfy_check.txt $logdir/cluvfy_check."$(date +%Y%m%d-%H%M%S)".txt
 fi
 
-cmd='su - $GRID_USER -c "ssh $node  \"$GRID_HOME/runcluvfy.sh stage -pre nodeadd -n $hostname -vip $vip_hostname\" | tee -a $logdir/cluvfy_check.txt"'
+#cmd='su - $GRID_USER -c "ssh $node  \"$GRID_HOME/runcluvfy.sh stage -pre nodeadd -n $hostname -vip $vip_hostname\" | tee -a $logdir/cluvfy_check.txt"'
```
Comment (Djelibeybi) [inline OracleDatabase/RAC/OracleRealApplicationClusters/dockerfiles/12.2.0.1/AddNode.sh]: Please remove commented out lines from scripts. There's no need to add these.

File: OracleDatabase/RAC/OracleRealApplicationClusters/dockerfiles/12.2.0.1/AddNode.sh [hunk 21]
```diff
@@ -575,50 +733,50 @@ local vip_hostname=$VIP_HOSTNAME
...
+#eval $cmd
+
+print_message "Nodes in the cluster ${CLUSTER_NODES[@]}"
+for cls_node in "${CLUSTER_NODES[@]}"; do
+print_message "ssh to the node $node and executing cvu checks on $cls_node"
+cmd='su - $GRID_USER -c "ssh $node  \"$GRID_HOME/runcluvfy.sh stage -pre nodeadd -n $cls_node\" | tee -a $logdir/cluvfy_check.txt"'
 eval $cmd
```
Comment (Djelibeybi) [inline OracleDatabase/RAC/OracleRealApplicationClusters/dockerfiles/12.2.0.1/AddNode.sh]: What's the benefit of setting `cmd` and then immediately `eval`'ing it? Why not just `eval` the string on line 753? This is true for all instances of the `cmd`/`eval` pairs later.

**Pull request merged**